core/domain/user_services.py

```diff
@@ -2654,6 +2654,10 @@ def get_checkpoints_in_order(
     Returns:
         list(str). List of all checkpoints of the exploration in sequential
         order.
+
+    Raises:
+        Exception. None destination can never be a checkpoint.
```

Comment:
Both of these conditions seem incorrect -- we have states with "try again" answer groups all the time and there's no issue with them being checkpoints. /cc @paprajapati9 in case he wants to leave any comments here.

Presumably we should just not add the destination to the BFS queue if the destination is None?

Reply:
> we have states with "try again" answer groups all the time and there's no issue with them being checkpoints

Yes agreed, but `"try again" answer groups` do not contain None dest, right? Because I have downloaded some exploration and I found that `try again answer group` points to its own state, for eg:

If we a state as 'home' and we have a try-again answer group then its destination will also point to 'Home'.

The only interaction that I have found which contains None destination is End-exploration and I think that one should not be a checkpoint ( if I'm not wrong ).

Also, from this comment (https://github.com/oppia/oppia/pull/15827#discussion_r945570610), it seems to me that every anwser_group's dest should exist and default_outcome's dest should also exist if default_outcome exists. 

Please correct me if I'm wrong!!

Reply:
Apologies -- on this one, you are correct! Thank you for checking.

I would just suggest rewording to "States with a null destination can never be a checkpoint." and similarly below. This is because checkpoints are states, and it's weird to say that a "destination can't be a checkpoint".